File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/Meal.java
package com.nlefler.glucloser.models;

import java.util.Date;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.Ignore;

/**
 * Created by Nathan Lefler on 12/11/14.
 */
public class Meal extends RealmObject implements BolusEvent, HasPlace {
    @Ignore
    public static final String ParseClassName = "Meal";

    @Ignore
    public static final String MealIdFieldName = "mealId";
    private String id;

    @Ignore
    public static final String MealDateFieldName = "date";
    private Date date;

    @Ignore
    public static final String PlaceFieldName = "place";
    private Place place;

    @Ignore
    public static final String CarbsFieldName = "carbs";
    private int carbs;

    @Ignore
    public static final String InsulinFieldName = "insulin";
    private float insulin;

    @Ignore
    public static final String BeforeSugarFieldName = "beforeSugar";
    private BloodSugar beforeSugar;

    @Ignore
    public static final String CorrectionFieldName = "correction";
    private boolean correction;

    @Ignore
    public static final String FoodListFieldName = "foods";
    private RealmList<Food> foods;

    public String getId() {
        return id;
    }

    public void setId(String id){
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
    public Place getPlace() {
        return place;
    }

    public void setPlace(Place place) {
        this.place = place;
    }

    public int getCarbs() {
        return carbs;
    }

    public void setCarbs(int carbs) {
        this.carbs = carbs;
    }

    public float getInsulin() {
        return insulin;
    }

    public void setInsulin(float insulin) {
        this.insulin = insulin;
    }

    public BloodSugar getBeforeSugar() {
        return beforeSugar;
    }

    public void setBeforeSugar(BloodSugar beforeSugar) {
        this.beforeSugar = beforeSugar;
    }

    public boolean isCorrection() {
        return this.correction;
    }

    public void setCorrection(boolean correction) {
        this.correction = correction;
    }

    @Override
    public RealmList<Food> getFoods() {
        return foods;
    }

    public void setFoods(RealmList<Food> foods) {
        this.foods = foods;
    }
}

File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/Meal.kt
package com.nlefler.glucloser.models

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.Ignore
import io.realm.annotations.RealmClass
import java.util.*

/**
 * Created by Nathan Lefler on 12/11/14.
 */
@RealmClass
public open class Meal : RealmObject(), BolusEvent, HasPlace {
    override open var id: String = UUID.randomUUID().toString()
    override open var date: Date = Date()
    override open var bolusPattern: BolusPattern? = null
    override open var carbs: Int = 0
    override open var insulin: Float = 0f
    override open var beforeSugar: BloodSugar? = null
    override open var isCorrection: Boolean = false
    override open var foods: RealmList<Food> = RealmList()
    override open var place: Place? = null

    companion object {
        @Ignore
        public val ParseClassName: String = "Meal"

        @Ignore
        public val MealIdFieldName: String = "mealId"

        @Ignore
        public val MealDateFieldName: String = "date"

        @Ignore
        public val PlaceFieldName: String = "place"

        @Ignore
        public val CarbsFieldName: String = "carbs"

        @Ignore
        public val InsulinFieldName: String = "insulin"

        @Ignore
        public val BeforeSugarFieldName: String = "beforeSugar"

        @Ignore
        public val CorrectionFieldName: String = "correction"

        @Ignore
        public val FoodListFieldName: String = "foods"

        @Ignore
        public val BolusPatternFieldName: String = "bolusPattern"
    }
}
